### PR TITLE
Rotor implementation

### PR DESCRIPTION
**Pull request by Jharopa**
applyRotor and reverseRotor implementation.

### CODE REVIEW

File: Week 4/Enigma - Java/src/main/java/Enigma.java [hunk 1]
```diff
@@ -86,13 +87,45 @@ private static String applyRotor(String message, String rotor){
                                                  
         Example input message: "Apples", rotor: HFPMRIBTJWYDXQLGUKOVSNAZEC
         Example outout: "HGGDRO" */
-        // TO DO - add your implementation
+        char[] alphabetArr = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".toCharArray();
+        char[] rotorArr = rotor.toCharArray();
+
+        HashMap<Character, Character> wiring = new HashMap<>();
```
Comment (heyitsgrace996) [inline Week 4/Enigma - Java/src/main/java/Enigma.java:93]: Good use of Hashmap for Rotor. We may consider a separate function to handle the mapping with a boolean flag for forward and reverse to be used in both functions to improve code re-use.

**Review by heyitsgrace996 — CHANGES_REQUESTED** [review]
Good use of Hashmap for Rotor. Maybe a good idea to consider a separate function to handle the mapping with a boolean flag for forward and reverse to be used in both functions.

File: Week 4/Enigma - Java/src/main/java/Enigma.java (hunk 1 above)
Comment (Jharopa) [reply Week 4/Enigma - Java/src/main/java/Enigma.java]: Implemented requested changes for you to check out.

**Review by heyitsgrace996 — APPROVED** [review]
Approved after change implemented.

**Review by matthew-c-atu — APPROVED** [review]
Looks good :)

**Pull request merged**